docs/core/deploying/native-aot/interop.md

```diff
@@ -13,12 +13,9 @@ While native code interop works similarly in Native AOT and non-AOT deployments,
 
 ## Direct P/Invoke calls
 
-The P/Invoke calls in AOT-compiled binaries are bound lazily at run time by default, for better compatibility. You can configure the AOT compiler to generate direct calls for selected P/Invoke methods that are bound during startup by the dynamic loader that comes with the operating system. The unmanaged libraries and entry points referenced via direct calls must always be available at run time, otherwise the native binary fails to start.
-
-The benefits of direct P/Invoke calls are:
+The benefit of direct P/Inokve calls is *better steady state performance*.
 
-- They have *better steady state performance*.
-- They make it possible to *link the unmanaged dependencies statically*.
```

Comment:
Did we run into problems with this?

Reply:
I think I might have misread this -- I took this statement to mean that it allows users to link unmanaged references statically, but does is it supposed to refer to the runtime's unmanaged dependencies? That is, the things we ship alongside Native AOT?

Reply:
Yeah, this is about linking native static libraries into NAOT executables so if we want to drop that from the docs, this is the thing to drop.

But there definitely are customers that are using this. It's generally not too problematic (we already don't control the libc version we're linking with, for example). Instead of trying to drop linking native static libraries into native AOT code from docs, I'd just add a warning similar to the one we have for building static libraries with NAOT to be linked into native apps. The problems are similar, managed static libraries have a bit more problems than managed apps linking native code statically.

Maybe I don't have enough context on what motivated this?

Cc @dotnet/ilc-contrib

Reply:
Context is that it is not currently clear what is supported with Native AOT and what isn't. My general rule is that we shouldn't be putting docs for unsupported scenarios on dot.net and instead move them to the repo docs.

Which of these scenarios are supported?

Reply:
Everything mentioned here is supported.

Reply:
OK, cool -- in that case I will keep all the content, just refactor.